### PR TITLE
another aproach to have a file-system-level, interprocess lock

### PR DESCRIPTION
**Pull request by wraetz**

### CODE REVIEW

**Review by christiam — CHANGES_REQUESTED** [review]
In BC_GCP_TOOLS.java:304 please delete dst_filename if there's an exception when downloading the file. Also on dowload() in BC_DATABASE_RDD_ENTRY.java, if the file was not properly downloaded, the application should exit rather than continue with a corrupt file in place.

File: pipeline/src/main/java/BC_FILE_LOCK.java [hunk 1]
```diff
@@ -27,66 +27,51 @@
...
 
+/**
+ * releaseing the lock: we delete the directory!
+ *
+*/
     public void release()
     {
-        if ( f.exists() ) f.delete();
+        try { f.delete(); }
+        catch( Exception e ) { }
```
Comment (boratyng) [inline pipeline/src/main/java/BC_FILE_LOCK.java]: We should at least report an exception. What if the directory cannot be deleted?

**Review by boratyng — CHANGES_REQUESTED** [review]

**Review by boratyng — APPROVED** [review]

**Review by christiam — CHANGES_REQUESTED** [review]
Thanks for addressing the deletion of the file. Can you please also terminate the application if the file download fails?

**Comment by wraetz** [comment]
Can you please stop blocking this pull-request, terminating the application in case of a failed download involves infrastructure changes across many classes. Let's make that a different, new pull-request.

**Comment by christiam** [comment]
Sure, that sounds good.

**Review by christiam — APPROVED** [review]

**Comment by wraetz** [comment]
error propagation from the worker to the master will be in the next pull-request

**Pull request merged**